### PR TITLE
[JENKINS-47510] RemoteLauncher.isUnix() is not overridden - always returns master OS

### PR DESCRIPTION
**Pull request by kuisathaverat**
See [JENKINS-47510](https://issues.jenkins-ci.org/browse/JENKINS-47510).

### Submitter checklist

- [X] JIRA issue is well described

@jtnord

### CODE REVIEW

File: src/main/java/hudson/plugins/sshslaves/RemoteLauncher.java [hunk 1]
```diff
@@ -140,4 +141,9 @@ private String makeCommandLine(List<String> cmd, FilePath _workDir) {
     public void kill(Map<String, String> modelEnvVars) throws IOException, InterruptedException {
         // no way to do this
     }
+
+  @Override
+  public boolean isUnix() {
+    return File.pathSeparatorChar == ':';
```
Comment (jvz) [inline src/main/java/hudson/plugins/sshslaves/RemoteLauncher.java]: Is this better than checking `!Functions.isWindows()`?

Comment (kuisathaverat) [reply src/main/java/hudson/plugins/sshslaves/RemoteLauncher.java]: I did not know about that function, it is better to use it.

File: src/main/java/hudson/plugins/sshslaves/RemoteLauncher.java [hunk 2]
```diff
@@ -140,4 +141,9 @@ private String makeCommandLine(List<String> cmd, FilePath _workDir) {
     public void kill(Map<String, String> modelEnvVars) throws IOException, InterruptedException {
         // no way to do this
     }
+
+  @Override
+  public boolean isUnix() {
+    return !Functions.isWindows();
```
Comment (jtnord) [inline src/main/java/hudson/plugins/sshslaves/RemoteLauncher.java]: does this actually run on the agent?
if not this returns the same s of the master which is substituting one big for a regression

Comment (kuisathaverat) [reply src/main/java/hudson/plugins/sshslaves/RemoteLauncher.java]: I've remembered why it was not merged.

**Pull request merged**